### PR TITLE
Add zuul based manila tempest tests

### PR DESCRIPTION
**Pull request by fmount**
Depends-On:  https://github.com/openstack-k8s-operators/ci-framework/pull/975

### CODE REVIEW

**Comment by fmount** [comment]
Testing via https://review.rdoproject.org/r/c/testproject/+/50787

**Comment by son-vyas** [comment]
you can't run manila tempest test like this, you will need to create shared resouces and if you want to restrict tests execution to manila then i think you can just change - 
```
cifmw_tempest_default_groups: &test_operator_list
  - manila-operator
```

**Comment by son-vyas** [comment]
I will update this pr with restricting tempest tests to manila only

**Comment by fmount** [comment]
recheck

**Comment by son-vyas** [comment]
https://logserver.rdoproject.org/28/51128/9/check/manila-operator-tempest/975960b/controller/ci-framework-data/tests/tempest/stestr_results.html

**Comment by fmount** [comment]
> https://logserver.rdoproject.org/28/51128/9/check/manila-operator-tempest/975960b/controller/ci-framework-data/tests/tempest/stestr_results.html

Yeah, this is cool and ready to go, however I'd like to keep it as non voting for now and use `Prow` jobs to determine if we can merge because of my comments on #197

**Comment by fmount** [comment]
recheck

**Comment by son-vyas** [comment]
recheck

**Review by marios — COMMENTED** [review]
seems reasonable as zuul job definitions go ;)

**Comment by son-vyas** [comment]
recheck

**Comment by son-vyas** [comment]
recheck

**Review by marios — COMMENTED** [review]
looks good but the new job is failing ... is that a consistent error? don't think this one in particular is related to this change

https://logserver.rdoproject.org/96/196/b47b03e5c2450cbbeee34b690a84f4b741826205/github-check/manila-operator-tempest/73b33cd/job-output.txt

2024-01-09 10:29:18.601233 | controller |
2024-01-09 10:29:18.601242 | controller | TASK [edpm_prepare : Wait for OpenStack controlplane to be deployed _raw_params=oc wait OpenStackControlPlane {{ cr_name }} --namespace={{ cifmw_install_yamls_defaults['NAMESPACE'] }} --for=condition=ready --timeout={{ cifmw_edpm_prepare_timeout }}m] ***
2024-01-09 10:29:18.601248 | controller | Tuesday 09 January 2024  10:29:18 -0500 (0:00:00.788)       0:12:33.556 *******
2024-01-09 10:59:18.958807 | controller | fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["oc", "wait", "OpenStackControlPlane", "openstack-galera-network-isolation", "--namespace=openstack", "--for=condition=ready", "--timeout=30m"], "delta": "0:30:00.145234", "end": "2024-01-09 10:59:18.930995", "msg": "non-zero return code", "rc": 1, "start": "2024-01-09 10:29:18.785761", "stderr": "error: timed out waiting for the condition on openstackcontrolplanes/openstack-galera-network-isolation", "stderr_lines": ["error: timed out waiting for the condition on openstackcontrolplanes/openstack-galera-network-isolation"], "stdout": "", "stdout_lines": []}

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
> looks good but the new job is failing ... is that a consistent error? don't think this one in particular is related to this change
> 
> https://logserver.rdoproject.org/96/196/b47b03e5c2450cbbeee34b690a84f4b741826205/github-check/manila-operator-tempest/73b33cd/job-output.txt
> 
> 2024-01-09 10:29:18.601233 | controller | 2024-01-09 10:29:18.601242 | controller | TASK [edpm_prepare : Wait for OpenStack controlplane to be deployed _raw_params=oc wait OpenStackControlPlane {{ cr_name }} --namespace={{ cifmw_install_yamls_defaults['NAMESPACE'] }} --for=condition=ready --timeout={{ cifmw_edpm_prepare_timeout }}m] *** 2024-01-09 10:29:18.601248 | controller | Tuesday 09 January 2024 10:29:18 -0500 (0:00:00.788) 0:12:33.556 ******* 2024-01-09 10:59:18.958807 | controller | fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["oc", "wait", "OpenStackControlPlane", "openstack-galera-network-isolation", "--namespace=openstack", "--for=condition=ready", "--timeout=30m"], "delta": "0:30:00.145234", "end": "2024-01-09 10:59:18.930995", "msg": "non-zero return code", "rc": 1, "start": "2024-01-09 10:29:18.785761", "stderr": "error: timed out waiting for the condition on openstackcontrolplanes/openstack-galera-network-isolation", "stderr_lines": ["error: timed out waiting for the condition on openstackcontrolplanes/openstack-galera-network-isolation"], "stdout": "", "stdout_lines": []}

rechecking to see how it behaves: it seems a flaky issue, let's confirm w/ the next run

**Comment by fmount** [comment]
I see weird things here [1], for instance:

```
Failed to pull image "nowhere/podified-antelope-centos9/openstack-ovn-nb-db-server:3ce2fae1cee941b77ea8cddaa07594c6
...
```

that can be applied for other containers (why the namespace is "nowhere"?), as well as swift failing to attach `pvc(s)` to pods.

[1] https://logserver.rdoproject.org/96/196/b47b03e5c2450cbbeee34b690a84f4b741826205/github-check/manila-operator-tempest/a7c5ff8/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/events.log

**Comment by fmount** [comment]
recheck

File: zuul.d/jobs.yaml [hunk 1]
```diff
@@ -10,3 +10,29 @@
...
         - manila
+
+- job:
+    name: manila-operator-tempest
+    parent: cifmw-base-multinode-tempest
+    attempts: 1
+    vars:
+      cifmw_edpm_prepare_update_os_containers: true
+      cifmw_set_openstack_containers_edpm: true
+      cifmw_set_openstack_containers_registry: "{{ content_provider_ip | default('nowhere') | trim }}"
```
Comment (fmount) [inline zuul.d/jobs.yaml:21]: So I definitely think there's something weird happening here.
If you look at [1]:

```
NAME                               READY   STATUS                  RESTARTS   AGE
pod/ceph                           1/1     Running                 0          41m
pod/dnsmasq-dns-665cc5dc88-5bflr   0/1     Init:ImagePullBackOff   0          31m
pod/dnsmasq-dns-c67bc7d76-9szvx    0/1     Init:ImagePullBackOff   0          31m
pod/memcached-0                    0/1     ImagePullBackOff        0          31m
pod/openstack-cell1-galera-0       0/1     Init:ImagePullBackOff   0          31m
pod/openstack-galera-0             0/1     Init:ImagePullBackOff   0          31m
pod/ovn-controller-p7h4w           0/3     ImagePullBackOff        0          31m
pod/ovsdbserver-nb-0               0/1     ImagePullBackOff        0          31m
pod/ovsdbserver-sb-0               0/1     ImagePullBackOff        0          31m
pod/rabbitmq-cell1-server-0        0/1     Init:ImagePullBackOff   0          31m
pod/rabbitmq-server-0              0/1     Init:ImagePullBackOff   0          31m
pod/swift-ring-rebalance-rcq2j     0/1     ImagePullBackOff        0          31m
pod/swift-storage-0                0/16    ContainerCreating       0          31m
```
most of the containers are failing (see [2] for more details), and this happens due to the fact that the `content_provider_id` is not there, and `nowhere`, the default, is selected [2].

@viroel @son-vyas wdyt about this? I see the `content-provider` job being green [3] but I'm not sure why the ip address is not resolved.

[1] https://logserver.rdoproject.org/96/196/b47b03e5c2450cbbeee34b690a84f4b741826205/github-check/manila-operator-tempest/e2c3ef2/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/all_resources.log

[2] https://logserver.rdoproject.org/96/196/b47b03e5c2450cbbeee34b690a84f4b741826205/github-check/manila-operator-tempest/e2c3ef2/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/events.log

[3] https://review.rdoproject.org/zuul/buildset/ae028675f5264536aeee19a9a63eac87

Comment (fmount) [reply zuul.d/jobs.yaml]: but from what I'm seeing we try to pull down containers from :80 instead of using :5001

File: zuul.d/jobs.yaml [hunk 2]
```diff
@@ -10,3 +10,29 @@
         CEPH_DATASIZE: "2Gi"
       cifmw_kuttl_tests_operator_list:
         - manila
+
+- job:
+    name: manila-operator-tempest
+    parent: cifmw-base-multinode-tempest
+    attempts: 1
+    vars:
+      cifmw_edpm_prepare_update_os_containers: true
```
Comment (viroel) [inline zuul.d/jobs.yaml:19]: cifmw_edpm_prepare_update_os_containers and cifmw_set_openstack_containers_* is used to replace the openstack services containers, but your content_provider job is building the manila-operator images. So, you don't need to add these vars here.

**Comment by fmount** [comment]
Closed in favor of https://github.com/openstack-k8s-operators/manila-operator/pull/267

**Pull request closed without merging**